### PR TITLE
Add appropriate message if empty bundle

### PR DESCRIPTION
**Pull request by sladyn98**
This PR fixes the app crashing if the bundle in the local cache is empty and instead adds an appropriate message.

### CODE REVIEW

File: frontend/src/components/PackageGeneration/Editor.js [hunk 1]
```diff
@@ -35,9 +35,25 @@ class editor extends React.Component {
...
+      this.setState({code: "Kindly generate or paste your configuration here"})
+    }
+
+    if (localStorage.getItem("packageConfigJSON")["bundle"] === undefined) {
+      this.setState({title: "No title Specified"})
+    } else {
+      this.setState({title: JSON.parse(localStorage.getItem("packageConfigJSON"))["bundle"]["title"]})
+    }
+
+    if (localStorage.getItem("packageConfigJSON")["bundle"] === undefined) {
```
Comment (martinda) [inline frontend/src/components/PackageGeneration/Editor.js:51]: This is exactly the same comparison as on line 45. Should it also the description? What if the bundle is null?

Comment (sladyn98) [reply frontend/src/components/PackageGeneration/Editor.js]: Yeah added an extra condition to fix it . Thanks

File: frontend/src/components/PackageGeneration/Editor.js [hunk 2]
```diff
@@ -35,9 +35,31 @@ class editor extends React.Component {
...
+
+    if (localStorage.getItem("packageConfigYAML")) {
+      this.setState({code: localStorage.getItem("packageConfigYAML")})
+    } else {
+      this.setState({code: "Kindly generate or paste your configuration here"})
+    }
+
+    if (!localStorage.getItem("packageConfigJSON")["bundle"] === undefined ) {
+      if (localStorage.getItem("packageConfigJSON")["bundle"]["title"] === undefined ) {
+        this.setState({title: "No title Specified"})
```
Comment (martinda) [inline frontend/src/components/PackageGeneration/Editor.js:47]: What do you expect the user to do under this condition?

Comment (sladyn98) [reply frontend/src/components/PackageGeneration/Editor.js:47]: The user cannot do anything. He cannot manually change the title unless he goes back and creates a new configuration

**Comment by sladyn98** [comment]
To test manually you need to have npm and mavnen installed
To start Front-end:
```
cd frontend/
npm start
```
To start backend
```
mvn spring-boot:run
```

File: frontend/src/components/PackageGeneration/Editor.js [hunk 3]
```diff
@@ -35,9 +35,31 @@ class editor extends React.Component {
...
+   }
+
+    if (!localStorage.getItem("packageConfigJSON")["bundle"] === undefined ) {
+      if (localStorage.getItem("packageConfigJSON")["bundle"]["description"] === undefined ) {
+        this.setState({description: "No description specified"})
+      }
+      else {
+        this.setState({description: JSON.parse(localStorage.getItem("packageConfigJSON"))["bundle"]["description"]})
+      }
+  }
```
Comment (kwhetstone) [inline frontend/src/components/PackageGeneration/Editor.js:61]: Should you also notes if there are issues if "bundle" isn't in the packageConfigJSON?

Comment (sladyn98) [reply frontend/src/components/PackageGeneration/Editor.js:61]: Should I add another condition to check if the bundle is present ?

**Comment by martinda** [comment]
I am having difficulties running this PR. I get these errors in the frontend:
```
Proxy error: Could not proxy request /favicon.ico from localhost:3000 to http://app-server:8080.
Proxy error: Could not proxy request /api/plugin/getPluginList from localhost:3000 to http://app-server:8080.
```

**Comment by sladyn98** [comment]
@kwhetstone @martinda Should we add more checks to this PR or is it good to go as is ?

File: frontend/src/components/PackageGeneration/Editor.js (hunk 3 above)
Comment (kwhetstone) [reply frontend/src/components/PackageGeneration/Editor.js:61]: Only if there's not a check earlier for if `JSON.parse(localStorage.getItem("packageConfigJSON"))["bundle"]` is missing.  If it's missing, then you can't do generation, but you should put something out to the console so you can trace where a problem happened.

Comment (sladyn98) [reply frontend/src/components/PackageGeneration/Editor.js:61]: These are just for the description for the cards so we do not really care if it present or not. If the generation errors out the backend comes back with the 404.

**Comment by martinda** [comment]
Hi @sladyn98 . I still get the errors I mentioned above. Those need to be addressed. This PR is behind master, so I suggest you start by rebasing it on master. Then write down the steps to test all aspects of this fix, then write a unit test.

**Comment by sladyn98** [comment]
Steps to test this PR:
a) Select plugins and generate a package.
b) Verify if the selected package name appears on the package generation page.
c) If no package is generated or the editor is empty verify the above messages appear 
    1) Title: No title specified
    2) Description: No Description specified

**Comment by sladyn98** [comment]
> Then write down the steps to test all aspects of this fix, then write a unit test.

For the testing of react we would need to choose a tool and then setup the architecture for it
https://reactjs.org/docs/testing.html#:~:text=There%20are%20a%20few%20ways,to%2Dend%E2%80%9D%20tests).

**Comment by martinda** [comment]
I get the infinite spinning wheel :-(

How I run it:
```
git fetch upstream refs/pull/119/head:refs/remotes/upstream/pull/119
git checkout pull/119
mvn clean
mvn spring-boot:run
```

In another terminal:
```
cd frontend
npm install && npm start
```

The errors:
```

Proxy error: Could not proxy request /favicon.ico from localhost:3000 to http://app-server:8080.
See https://nodejs.org/api/errors.html#errors_common_system_errors for more information (ENOTFOUND).

Proxy error: Could not proxy request /api/plugin/getPluginList from localhost:3000 to http://app-server:8080.
See https://nodejs.org/api/errors.html#errors_common_system_errors for more information (ENOTFOUND).

Proxy error: Could not proxy request /favicon.ico from localhost:3000 to http://app-server:8080.
See https://nodejs.org/api/errors.html#errors_common_system_errors for more information (ENOTFOUND).
```

**Comment by sladyn98** [comment]
@martinda you can only run this pr using docker since multiple environments is not yet supported.we need to merge that first to get this working using npm and spring Boot

**Comment by martinda** [comment]
Ok, so with docker "it runs" but I am still clueless with regards to testing this. What sequence of buttons should I click in the app? Should I expect error messages in the browser?

**Comment by sladyn98** [comment]
> Steps to test this PR:
> a) Select plugins and generate a package.
> b) Verify if the selected package name appears on the package generation page.
> c) If no package is generated or the editor is empty verify the above messages appear
> 1) Title: No title specified
> 2) Description: No Description specified

@martinda

**Pull request merged**